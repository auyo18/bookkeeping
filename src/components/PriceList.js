import React from 'react'
import PropTypes from 'prop-types'
import {List, Row, Col} from 'antd'
import IosCreate from 'react-ionicons/lib/IosCreate'
import IosTrash from 'react-ionicons/lib/IosTrash'

const getIcon = name => {
  const Icon = require(`react-ionicons/lib/${name}.js`)
  return <Icon onClick={() => alert('Hi!')} fontSize="40px" color="#434343"
               style={{backgroundColor: '#ffda44', borderRadius: '50%', padding: '5px'}}/>
}

const PriceList = ({items, onModifyItem, onDeleteItem}) => (
  <List
    bordered
    itemLayout="vertical"
    dataSource={items}
    renderItem={item => (
      <List.Item>
        <Row type="flex" justify="space-between" align="middle">
          <Col className="gutter-row" span={4}>
            <div className="gutter-box">
              {
                getIcon(item.category.iconName)
              }
            </div>
          </Col>
          <Col className="gutter-row" span={6}>
            <div className="gutter-box">
              {item.title}
            </div>
          </Col>
          <Col className="gutter-row" span={4}>
            <div className="gutter-box">
              <span>{item.category.type === 'income' ? '' : '-'}{item.price}</span>
            </div>
          </Col>
          <Col className="gutter-row" span={4}>
            <div className="gutter-box">
              <span>{item.data}</span>
            </div>
          </Col>
          <Col className="gutter-row" span={3}>
            <div className="gutter-box">
              <IosCreate onClick={() => onModifyItem(item)} fontSize="40px" color="white"
                         style={{backgroundColor: '#434343', borderRadius: '50%', padding: '5px'}}/>
            </div>
          </Col>
          <Col className="gutter-row" span={3}>
            <div className="gutter-box">
              <IosTrash onClick={() => onDeleteItem(item)} fontSize="40px" color="white"
                        style={{backgroundColor: '#f5222d', borderRadius: '50%', padding: '5px'}}/>
            </div>
          </Col>
        </Row>
      </List.Item>
    )}
  />
)

PriceList.propTypes = {
  items: PropTypes.array.isRequired,
  onModifyItem: PropTypes.func.isRequired,
  onDeleteItem: PropTypes.func.isRequired
}

export default PriceList
