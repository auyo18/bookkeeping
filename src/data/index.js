export const items = [
  {
    "id": 1,
    "title": "去云南旅游",
    "price": 6600,
    "date": "2019-05-12",
    "cid": 1
  },
  {
    "id": 2,
    "title": "超市购物",
    "price": 230,
    "date": "2019-05-26",
    "cid": 2
  },
  {
    "id": 3,
    "title": "工资",
    "price": 8000,
    "date": "2019-05-15",
    "cid": 3
  }
]

export const category = [
  {
    "id": 1,
    "name": "旅游",
    "type": "outcome"
  },
  {
    "id": 2,
    "name": "购物",
    "type": "outcome"
  },
  {
    "id": 3,
    "name": "工资",
    "type": "income"
  }
]
