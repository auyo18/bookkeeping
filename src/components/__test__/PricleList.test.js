import React from 'react'
import {Icon} from 'antd'
import {render} from 'enzyme'
import toJson from 'enzyme-to-json'
import PriceList from '../PriceList'
import {items, category} from "../../data"

items.forEach(item => {
  category.forEach(cg => {
    if (item.cid === cg.id) {
      item.category = cg
    }
  })
})

const props = {
  items,
  onModifyItem: () => {
  },
  onDeleteItem: () => {
  }
}

let wrapper
describe('test PriceList component', () => {
  beforeEach(() => {
    wrapper = render(<PriceList {...props}/>)
  })
  it('should render the component to match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('should render correct price items length', () => {
    expect(wrapper.find('.ant-list-item').length).toEqual(items.length)
  })
})
