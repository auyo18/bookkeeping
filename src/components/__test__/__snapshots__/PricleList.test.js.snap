// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PriceList component should render the component to match snapshot 1`] = `
<div
  class="ant-list ant-list-vertical ant-list-split ant-list-bordered"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <ul
        class="ant-list-items"
      >
        <li
          class="ant-list-item ant-list-item-no-flex"
        >
          <div
            class="ant-row-flex ant-row-flex-space-between ant-row-flex-middle"
          >
            <div
              class="ant-col ant-col-4 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <i
                  aria-label="icon: gitlab"
                  class="anticon anticon-gitlab"
                  style="background-color:#ffda44;padding:10px;font-size:30px;border-radius:50%"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="gitlab"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M913.9 552.2L805 181.4v-.1c-7.6-22.9-25.7-36.5-48.3-36.5-23.4 0-42.5 13.5-49.7 35.2l-71.4 213H388.8l-71.4-213c-7.2-21.7-26.3-35.2-49.7-35.2-23.1 0-42.5 14.8-48.4 36.6L110.5 552.2c-4.4 14.7 1.2 31.4 13.5 40.7l368.5 276.4c2.6 3.6 6.2 6.3 10.4 7.8l8.6 6.4 8.5-6.4c4.9-1.7 9-4.7 11.9-8.9l368.4-275.4c12.4-9.2 18-25.9 13.6-40.6zM751.7 193.4c1-1.8 2.9-1.9 3.5-1.9 1.1 0 2.5.3 3.4 3L818 394.3H684.5l67.2-200.9zm-487.4 1c.9-2.6 2.3-2.9 3.4-2.9 2.7 0 2.9.1 3.4 1.7l67.3 201.2H206.5l57.8-200zM158.8 558.7l28.2-97.3 202.4 270.2-230.6-172.9zm73.9-116.4h122.1l90.8 284.3-212.9-284.3zM512.9 776L405.7 442.3H620L512.9 776zm157.9-333.7h119.5L580 723.1l90.8-280.8zm-40.7 293.9l207.3-276.7 29.5 99.2-236.8 177.5z"
                    />
                  </svg>
                </i>
              </div>
            </div>
            <div
              class="ant-col ant-col-6 gutter-row"
            >
              <div
                class="gutter-box"
              >
                去云南旅游
              </div>
            </div>
            <div
              class="ant-col ant-col-4 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <span>
                  -6600
                </span>
              </div>
            </div>
            <div
              class="ant-col ant-col-4 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <span>
                  2019-05-12
                </span>
              </div>
            </div>
            <div
              class="ant-col ant-col-3 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <button
                  class="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg ant-btn-icon-only"
                  type="button"
                >
                  <i
                    aria-label="icon: edit"
                    class="anticon anticon-edit"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="edit"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                      />
                    </svg>
                  </i>
                </button>
              </div>
            </div>
            <div
              class="ant-col ant-col-3 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <button
                  class="ant-btn ant-btn-danger ant-btn-circle ant-btn-lg ant-btn-icon-only"
                  type="button"
                >
                  <i
                    aria-label="icon: delete"
                    class="anticon anticon-delete"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="delete"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                      />
                    </svg>
                  </i>
                </button>
              </div>
            </div>
          </div>
        </li>
        <li
          class="ant-list-item ant-list-item-no-flex"
        >
          <div
            class="ant-row-flex ant-row-flex-space-between ant-row-flex-middle"
          >
            <div
              class="ant-col ant-col-4 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <i
                  aria-label="icon: gitlab"
                  class="anticon anticon-gitlab"
                  style="background-color:#ffda44;padding:10px;font-size:30px;border-radius:50%"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="gitlab"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M913.9 552.2L805 181.4v-.1c-7.6-22.9-25.7-36.5-48.3-36.5-23.4 0-42.5 13.5-49.7 35.2l-71.4 213H388.8l-71.4-213c-7.2-21.7-26.3-35.2-49.7-35.2-23.1 0-42.5 14.8-48.4 36.6L110.5 552.2c-4.4 14.7 1.2 31.4 13.5 40.7l368.5 276.4c2.6 3.6 6.2 6.3 10.4 7.8l8.6 6.4 8.5-6.4c4.9-1.7 9-4.7 11.9-8.9l368.4-275.4c12.4-9.2 18-25.9 13.6-40.6zM751.7 193.4c1-1.8 2.9-1.9 3.5-1.9 1.1 0 2.5.3 3.4 3L818 394.3H684.5l67.2-200.9zm-487.4 1c.9-2.6 2.3-2.9 3.4-2.9 2.7 0 2.9.1 3.4 1.7l67.3 201.2H206.5l57.8-200zM158.8 558.7l28.2-97.3 202.4 270.2-230.6-172.9zm73.9-116.4h122.1l90.8 284.3-212.9-284.3zM512.9 776L405.7 442.3H620L512.9 776zm157.9-333.7h119.5L580 723.1l90.8-280.8zm-40.7 293.9l207.3-276.7 29.5 99.2-236.8 177.5z"
                    />
                  </svg>
                </i>
              </div>
            </div>
            <div
              class="ant-col ant-col-6 gutter-row"
            >
              <div
                class="gutter-box"
              >
                超市购物
              </div>
            </div>
            <div
              class="ant-col ant-col-4 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <span>
                  -230
                </span>
              </div>
            </div>
            <div
              class="ant-col ant-col-4 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <span>
                  2019-05-26
                </span>
              </div>
            </div>
            <div
              class="ant-col ant-col-3 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <button
                  class="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg ant-btn-icon-only"
                  type="button"
                >
                  <i
                    aria-label="icon: edit"
                    class="anticon anticon-edit"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="edit"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                      />
                    </svg>
                  </i>
                </button>
              </div>
            </div>
            <div
              class="ant-col ant-col-3 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <button
                  class="ant-btn ant-btn-danger ant-btn-circle ant-btn-lg ant-btn-icon-only"
                  type="button"
                >
                  <i
                    aria-label="icon: delete"
                    class="anticon anticon-delete"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="delete"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                      />
                    </svg>
                  </i>
                </button>
              </div>
            </div>
          </div>
        </li>
        <li
          class="ant-list-item ant-list-item-no-flex"
        >
          <div
            class="ant-row-flex ant-row-flex-space-between ant-row-flex-middle"
          >
            <div
              class="ant-col ant-col-4 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <i
                  aria-label="icon: gitlab"
                  class="anticon anticon-gitlab"
                  style="background-color:#ffda44;padding:10px;font-size:30px;border-radius:50%"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="gitlab"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M913.9 552.2L805 181.4v-.1c-7.6-22.9-25.7-36.5-48.3-36.5-23.4 0-42.5 13.5-49.7 35.2l-71.4 213H388.8l-71.4-213c-7.2-21.7-26.3-35.2-49.7-35.2-23.1 0-42.5 14.8-48.4 36.6L110.5 552.2c-4.4 14.7 1.2 31.4 13.5 40.7l368.5 276.4c2.6 3.6 6.2 6.3 10.4 7.8l8.6 6.4 8.5-6.4c4.9-1.7 9-4.7 11.9-8.9l368.4-275.4c12.4-9.2 18-25.9 13.6-40.6zM751.7 193.4c1-1.8 2.9-1.9 3.5-1.9 1.1 0 2.5.3 3.4 3L818 394.3H684.5l67.2-200.9zm-487.4 1c.9-2.6 2.3-2.9 3.4-2.9 2.7 0 2.9.1 3.4 1.7l67.3 201.2H206.5l57.8-200zM158.8 558.7l28.2-97.3 202.4 270.2-230.6-172.9zm73.9-116.4h122.1l90.8 284.3-212.9-284.3zM512.9 776L405.7 442.3H620L512.9 776zm157.9-333.7h119.5L580 723.1l90.8-280.8zm-40.7 293.9l207.3-276.7 29.5 99.2-236.8 177.5z"
                    />
                  </svg>
                </i>
              </div>
            </div>
            <div
              class="ant-col ant-col-6 gutter-row"
            >
              <div
                class="gutter-box"
              >
                工资
              </div>
            </div>
            <div
              class="ant-col ant-col-4 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <span>
                  8000
                </span>
              </div>
            </div>
            <div
              class="ant-col ant-col-4 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <span>
                  2019-05-15
                </span>
              </div>
            </div>
            <div
              class="ant-col ant-col-3 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <button
                  class="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg ant-btn-icon-only"
                  type="button"
                >
                  <i
                    aria-label="icon: edit"
                    class="anticon anticon-edit"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="edit"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                      />
                    </svg>
                  </i>
                </button>
              </div>
            </div>
            <div
              class="ant-col ant-col-3 gutter-row"
            >
              <div
                class="gutter-box"
              >
                <button
                  class="ant-btn ant-btn-danger ant-btn-circle ant-btn-lg ant-btn-icon-only"
                  type="button"
                >
                  <i
                    aria-label="icon: delete"
                    class="anticon anticon-delete"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="delete"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                      />
                    </svg>
                  </i>
                </button>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
